### PR TITLE
Resetting no matter where the video currently is

### PR DESCRIPTION
**Pull request by rudibravo**
Reapplying that fix with some extra call to `load` and a way to detect and log the issue if it reappears.
The extra call to `load` improved a lot the situation on Chrome 59 and the timer prevent the videos from hanging forever on the loading/pending state.

@Rise-Vision/content @stulees please review

### CODE REVIEW

File: src/widget/player-vjs.js [hunk 1]
```diff
@@ -116,6 +117,28 @@ RiseVision.Video.PlayerVJS = function PlayerVJS( params, mode ) {
     _playerInstance.on( "ended", _onEnded );
     _playerInstance.on( "error", _onError );
     _playerInstance.on( "loadedmetadata", _onLoadedMetaData );
+    _playerInstance.on( "loadstart", _onLoadStart );
+    _playerInstance.on( "canplay", _onCanPlay );
```
Comment (stulees) [inline src/widget/player-vjs.js]: @rudibravo When does VJS fire `canplay`?

Comment (rudibravo) [reply src/widget/player-vjs.js]: @stulees it fires when the video can actually start playing, but before the whole video can be played without buffer (canplaythrought).

During the loading process of an audio/video, the following events occur, in this order:

loadstart
durationchange
loadedmetadata
loadeddata
progress
canplay
canplaythrough

File: src/widget/player-vjs.js [hunk 2]
```diff
@@ -116,6 +117,28 @@ RiseVision.Video.PlayerVJS = function PlayerVJS( params, mode ) {
...
+
+  function _onCanPlay() {
+    clearTimeout( _loadTimer );
+  }
+
+  function _onLoadStart() {
+    clearTimeout( _loadTimer );
+
+    _loadTimer = setTimeout( function logError() {
+      if ( !_playerInstance.readyState() ) {
```
Comment (stulees) [inline src/widget/player-vjs.js]: @rudibravo Have you been able to replicate this getting caught here? `readyState()` won't always have an applicable status provided?

File: src/widget/player-vjs.js (hunk 1 above)
Comment (stulees) [reply src/widget/player-vjs.js]: @rudibravo Okay thanks

File: src/widget/player-vjs.js (hunk 2 above)
Comment (rudibravo) [reply src/widget/player-vjs.js]: @stulees Yes, this is where I'm checking if the request is pending forever. If the player instance still have nothing loaded after 10 seconds this state will be 0 and I can assume the loading failed and restart.

0 = HAVE_NOTHING - no information whether or not the audio/video is ready
1 = HAVE_METADATA - metadata for the audio/video is ready
2 = HAVE_CURRENT_DATA - data for the current playback position is available, but not enough data to play next frame/millisecond
3 = HAVE_FUTURE_DATA - data for the current and at least the next frame is available
4 = HAVE_ENOUGH_DATA - enough data available to start playing

Comment (stulees) [reply src/widget/player-vjs.js]: @rudibravo Okay thanks

**Comment by jungeunyoon** [comment]
[widget-video -- Issue 201 [3]](https://trello.com/c/6O1u8vxS/5121-widget-video-issue-201-3)

**Pull request merged**